// react-native-password-strength-meter-bar/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _Bar = _interopRequireDefault(require("react-native-progress/Bar"));

var _zxcvbn = _interopRequireDefault(require("zxcvbn"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const PasswordStrengthMeterBar = ({
  password,
  radius = 4,
  height = 8,
  showStrenghtText = true,
  unfilledColor = '#F0F0F0'
}) => {
  const [result, setResult] = (0, _react.useState)(0);
  const [testedResult, setTestedResult] = (0, _react.useState)(0);
  const [color, setColor] = (0, _react.useState)('#F25F5C');
  const [label, setLabel] = (0, _react.useState)('');
  const [firstMount, setFirstMount] = (0, _react.useState)(0);

  const calculaterPercentage = value => {
    switch (value) {
      case 0:
        return 0;

      case 1:
        return 0.25;

      case 2:
        return 0.5;

      case 3:
        return 0.75;

      case 4:
        return 1;

      default:
        return 0;
    }
  };

  const calculateBarColor = value => {
    switch (value) {
      case 0:
        return '#F25F5C';

      case 1:
        return '#F25F5C';

      case 2:
        return '#FFE066';

      case 3:
        return '#247BA0';

      case 4:
        return '#70C1B3';

      default:
        return '#F25F5C';
    }
  };

  const calculateLabel = value => {
    switch (value) {
      case 0:
        return 'Weak';

      case 1:
        return 'Weak';

      case 2:
        return 'Fair';

      case 3:
        return 'Good';

      case 4:
        return 'Strong';

      default:
        return 'Weak';
    }
  };

  (0, _react.useEffect)(() => {
    setTestedResult((0, _zxcvbn.default)(password).score);
    setResult(calculaterPercentage(testedResult));
    setLabel(calculateLabel(testedResult));
    setColor(calculateBarColor(testedResult));
    setFirstMount(firstMount + 1);
    return () => {
      setFirstMount(0);
    }; // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [password]);
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: {
      marginTop: 5
    }
  }, testedResult > 0 || firstMount > 1 ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Bar.default, {
    progress: result,
    width: null,
    height: height,
    color: color,
    unfilledColor: unfilledColor,
    borderColor: "transparent",
    borderRadius: radius
  }), showStrenghtText && /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: {
      color
    }
  }, label)) : /*#__PURE__*/_react.default.createElement(_reactNative.View, null));
};

var _default = PasswordStrengthMeterBar;
exports.default = _default;
